KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift

```diff
@@ -202,41 +300,58 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
         }
     }
     
-    private func updateViewFromViewModel() {
+    private func updateViewVisibilityFromViewModel() {
         let canEarn = viewModel?.canEarn ?? false
         let canBurn = viewModel?.canBurn ?? false
         
         view?.updateLoyaltyFeatures(showEarnRelatedUI: canEarn, showBurnRelatedUI: canBurn)
     }
     
+    private func updateViewVisibilityFromViewModelForBurnMode() {
+        let canBurn = viewModel?.canBurn ?? false
+        
+        view?.updateLoyaltyFeatures(showEarnRelatedUI: true, showBurnRelatedUI: canBurn)
+    }
+    
+    private func updateViewForGetEarnAmountError() {
+        let canBurn = viewModel?.canBurn ?? false
+        
+        view?.updateLoyaltyFeatures(showEarnRelatedUI: false, showBurnRelatedUI: canBurn)
+    }
+    
     private func hideLoyaltyComponent() {
         view?.updateLoyaltyFeatures(showEarnRelatedUI: false, showBurnRelatedUI: false)
     }
     
-    // TODO: remove this method and the related enum, var, etc. They are only for testing the UI in burn mode while under development
-    private func showcaseBurnMode() {
-        guard currentMode == .burn else {
-            return
-        }
-        
-        switch showcaseBurnModeType {
-        case .valid:
-            updateViewFromViewModel()
-            showcaseBurnModeType = .errorInsufficientBalance
-            
-        case .errorInsufficientBalance:
-            view?.showError(withMessage: UITexts.Errors.insufficientBalanceForLoyaltyBurning)
-            view?.updateLoyaltyFeatures(showEarnRelatedUI: true, showBurnRelatedUI: viewModel?.canBurn ?? false)
-            showcaseBurnModeType = .errorUnsupportedCurrency
-            
-        case .errorUnsupportedCurrency:
-            view?.showError(withMessage: UITexts.Errors.unsupportedCurrency)
-            view?.updateLoyaltyFeatures(showEarnRelatedUI: true, showBurnRelatedUI: viewModel?.canBurn ?? false)
-            showcaseBurnModeType = .valid
+    private func handleEarnPointsCallError() {
+        // TODO: Once the slug is added to the error returned from the server, uncomment and update the snippet below
+        // to show the unsupported currency error only when it is indeed the case
+        // and delete the call to updateUIWithError on the next line
+        if currentMode == .earn, (viewModel?.canEarn ?? false) {
+            updateUIWithError(message: UITexts.Errors.unsupportedCurrency)
         }
+//                switch error?.type {
```

Comment:
Do we need to keep the commented code?

Reply:
it's part of a TODO for when the slug issue will be resolved (hopefully soon)